dev-client/src/model/permissions/permissions.test.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {User} from 'terraso-client-shared/account/accountSlice';
+import {Project, ProjectRole} from 'terraso-client-shared/project/projectSlice';
+
+import {userHasProjectRole} from 'terraso-mobile-client/model/permissions/permissions';
+
+const emptyUser = null;
+const emptyRoles: ProjectRole[] = [];
+
+const sampleUser: User = {
+  id: '1',
+  email: 'email@example.com',
+  firstName: 'Test',
+  lastName: 'Test',
+  profileImage: '',
+  preferences: {
+    group_notifications: 'true',
+    story_map_notifications: 'true',
+    language: 'en-US',
+  },
+};
+
+const emptyProject: Project = {
+  id: '3',
```

Comment:
Can we condense these project objects to a helper factory method? It's difficult to parse many large objects with fields that aren't all relevant to the test, I'd rather see like `const sampleProject = projectWithMemberships(...);` or something like that.